src/DynamoUtilities/CertificateVerification.cs

```diff
@@ -3,12 +3,194 @@
+        {
+            Success = 0,
+            ProviderUnknown = 0x800b0001,           // Trust provider is not recognized on this system
+            ActionUnknown = 0x800b0002,         // Trust provider does not support the specified action
+            SubjectFormUnknown = 0x800b0003,        // Trust provider does not support the form specified for the subject
+            SubjectNotTrusted = 0x800b0004,         // Subject failed the specified verification action
+            FileNotSigned = 0x800B0100,         // TRUST_E_NOSIGNATURE - File was not signed
+            SubjectExplicitlyDistrusted = 0x800B0111,   // Signer's certificate is in the Untrusted Publishers store
+            SignatureOrFileCorrupt = 0x80096010,    // TRUST_E_BAD_DIGEST - file was probably corrupt
+            SubjectCertExpired = 0x800B0101,        // CERT_E_EXPIRED - Signer's certificate was expired
+            SubjectCertificateRevoked = 0x800B010C,     // CERT_E_REVOKED Subject's certificate was revoked
+            UntrustedRoot = 0x800B0109          // CERT_E_UNTRUSTEDROOT - A certification chain processed correctly but terminated in a root certificate that is not trusted by the trust provider.
+        }
+
+        public class WinTrust
+        {
+            private static readonly IntPtr INVALID_HANDLE_VALUE = new IntPtr(-1);
+            // GUID of the action to perform
+            private const string WINTRUST_ACTION_GENERIC_VERIFY_V2 = "{00AAC56B-CD44-11d0-8CC2-00C04FC295EE}";
+
+            [DllImport("wintrust.dll", ExactSpelling = true, SetLastError = false, CharSet = CharSet.Unicode)]
+            static extern WinVerifyTrustResult WinVerifyTrust(
+                [In] IntPtr hwnd,
+                [In] [MarshalAs(UnmanagedType.LPStruct)] Guid pgActionID,
+                [In] WinTrustData pWVTData
+            );
+
+            // call WinTrust.WinVerifyTrust() to check embedded file signature
+            public static bool VerifyEmbeddedSignature(string fileName)
+            {
+                WinTrustFileInfo winTrustFileInfo = null;
+                WinTrustData winTrustData = null;
+
+                try
+                {
+                    winTrustFileInfo = new WinTrustFileInfo(fileName);
+                    winTrustData = new WinTrustData(winTrustFileInfo);
+                    Guid guidAction = new Guid(WINTRUST_ACTION_GENERIC_VERIFY_V2);
+                    WinVerifyTrustResult result = WinVerifyTrust(INVALID_HANDLE_VALUE, guidAction, winTrustData);
+                    bool ret = (result == WinVerifyTrustResult.Success);
+                    return ret;
+                }
+                finally
+                {
+                    // free the locally-held unmanaged memory in the data structures
+                    if (winTrustFileInfo != null) winTrustFileInfo.Dispose();
+                    if (winTrustData != null) winTrustData.Dispose();
+                }
+            }
+            private WinTrust() { }
```

Comment:
why is this here?

Reply:
This does not seem to be necessary

Reply:
The purpose is to keep other classes from being able to construct instances of this otherwise static class.